From fc1a2fc52799e73396798f17df70e0383f4b95b0 Mon Sep 17 00:00:00 2001
From: Arne Coucheron <arco68@gmail.com>
Date: Tue, 29 Oct 2019 00:26:15 +0100
Subject: [PATCH] 2G wants proper signal strength too

Change-Id: Ibe3acd8e214075199127b4c114a222b26aaf8498
(cherry picked from commit 2b6d45d435c6ebc98cafb9820a44fb50e2c4dd16)
(cherry picked from commit 3949202159e1839befd528d644a081f17a852dd2)
---
 src/java/com/android/internal/telephony/RIL.java | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/java/com/android/internal/telephony/RIL.java b/src/java/com/android/internal/telephony/RIL.java
index 8429c49db5..88164053f3 100644
--- a/src/java/com/android/internal/telephony/RIL.java
+++ b/src/java/com/android/internal/telephony/RIL.java
@@ -7460,11 +7460,14 @@ public class RIL extends BaseCommands implements CommandsInterface {
             }
         }
         switch (voiceRat) {
+            case ServiceState.RIL_RADIO_TECHNOLOGY_GPRS: /* fallthrough */
+            case ServiceState.RIL_RADIO_TECHNOLOGY_EDGE: /* fallthrough */
             case ServiceState.RIL_RADIO_TECHNOLOGY_UMTS: /* fallthrough */
             case ServiceState.RIL_RADIO_TECHNOLOGY_HSDPA: /* fallthrough */
             case ServiceState.RIL_RADIO_TECHNOLOGY_HSUPA: /* fallthrough */
             case ServiceState.RIL_RADIO_TECHNOLOGY_HSPA: /* fallthrough */
             case ServiceState.RIL_RADIO_TECHNOLOGY_HSPAP: /* fallthrough */
+            case ServiceState.RIL_RADIO_TECHNOLOGY_GSM: /* fallthrough */
                 break;
             default:
                 // If we are not currently on WCDMA/HSPA, then we don't need to do a fixup.
